homeassistant/components/wmspro/cover.py

```diff
@@ -0,0 +1,78 @@
+"""Support for covers connected with WMS WebControl pro."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from wmspro.const import (
+    WMS_WebControl_pro_API_actionDescription,
+    WMS_WebControl_pro_API_actionType,
+)
+
+from homeassistant.components.cover import ATTR_POSITION, CoverDeviceClass, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .generic_entity import WebControlProGenericEntity
+
+SCAN_INTERVAL = timedelta(seconds=5)
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the WMS based covers from a config entry."""
+    hub = config_entry.runtime_data
+
+    entities: list[WebControlProGenericEntity] = []
+    for dest in hub.dests.values():
+        await dest.refresh()
```

Comment:
Why do we refresh here?

Reply:
Without a refresh here the destinations actions aka device type and capabilities would be unknown/unset.

Reply:
Should we move this to `__init__.py`? Because if you're going to add a new platform in the future, all the dests would be refreshed again. This way we only refresh once before startup